x-pack/plugins/enterprise_search/common/types/engines.ts

```diff
@@ -20,11 +20,14 @@ export interface EnterpriseSearchEngine {
 }
 
 export interface EnterpriseSearchEngineDetails {
-  indices: EnterpriseSearchEngineIndex[];
+  indices: string[];
```

Comment:
indices are of type string[] from ES but the hydrated indices is of type EnterpriseSearchEngineIndex[]. Hence added `EnterpriseSearchEngineDetailsResponse`

Reply:
This type was only needed because the get engine & list engines results differed. since this is returning `indices: string[];` now we can just use `EnterpriseSearchEngine` as the types are the same.